test/e2e/singlecluster/statefulset_test.go

```diff
@@ -0,0 +1,137 @@
+					WithinClusterQueue: kueue.PreemptionPolicyLowerPriority,
+				}).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, cq)).To(gomega.Succeed())
+			lq = testing.MakeLocalQueue(localQueueName, ns.Name).ClusterQueue(cq.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, lq)).To(gomega.Succeed())
+		})
+		ginkgo.AfterEach(func() {
+			gomega.Expect(util.DeleteAllPodsInNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+			util.ExpectObjectToBeDeleted(ctx, k8sClient, cq, true)
+		})
+
+		ginkgo.It("should admit group that fits", func() {
+			statefulSet := statefulsettesting.MakeStatefulSet("sf", ns.Name).
+				Image(util.E2eTestSleepImage, []string{"10m"}).
+				Request(corev1.ResourceCPU, "1").
+				Limit(corev1.ResourceCPU, "1").
+				Replicas(3).
+				Queue(lq.Name).
+				Obj()
+			wlLookupKey := types.NamespacedName{Name: statefulset.GetWorkloadName(statefulSet.Name, statefulSet.UID), Namespace: ns.Name}
+			gomega.Expect(k8sClient.Create(ctx, statefulSet)).To(gomega.Succeed())
+
+			gomega.Eventually(func(g gomega.Gomega) {
+				createdStatefulSet := &appsv1.StatefulSet{}
+				g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(statefulSet), createdStatefulSet)).
+					To(gomega.Succeed())
+				g.Expect(createdStatefulSet.Status.ReadyReplicas).To(gomega.Equal(int32(3)))
+			}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+			createdWorkload := &kueue.Workload{}
+			gomega.Expect(k8sClient.Get(ctx, wlLookupKey, createdWorkload)).To(gomega.Succeed())
+
+			ginkgo.By("Creating potentially conflicting stateful-set", func() {
+				conflictingStatefulSet := statefulsettesting.MakeStatefulSet("sf-conflict", ns.Name).
+					Image(util.E2eTestSleepImage, []string{"10m"}).
+					Request(corev1.ResourceCPU, "0.1").
+					Limit(corev1.ResourceCPU, "0.1").
+					Replicas(1).
+					Queue(lq.Name).
+					Obj()
+				conflictingWlLookupKey := types.NamespacedName{
+					Name:      statefulset.GetWorkloadName(conflictingStatefulSet.Name, conflictingStatefulSet.UID),
+					Namespace: ns.Name,
+				}
+				gomega.Expect(k8sClient.Create(ctx, conflictingStatefulSet)).To(gomega.Succeed())
+				gomega.Eventually(func(g gomega.Gomega) {
+					createdStatefulSet := &appsv1.StatefulSet{}
+					g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(conflictingStatefulSet), createdStatefulSet)).
+						To(gomega.Succeed())
+					g.Expect(createdStatefulSet.Status.ReadyReplicas).To(gomega.Equal(int32(1)))
+				}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+				conflictingWorkload := &kueue.Workload{}
+				gomega.Expect(k8sClient.Get(ctx, conflictingWlLookupKey, conflictingWorkload)).To(gomega.Succeed())
+				gomega.Expect(createdWorkload.Name).ToNot(gomega.Equal(conflictingWorkload.Name))
+				util.ExpectObjectToBeDeleted(ctx, k8sClient, conflictingStatefulSet, true)
+				util.ExpectObjectToBeDeleted(ctx, k8sClient, conflictingWorkload, false)
+			})
+
+			util.ExpectObjectToBeDeleted(ctx, k8sClient, statefulSet, true)
```

Comment:
I think it will be better to put this kind of calls in a defer just after the object's creation